lib/driver.rb

```diff
@@ -39,9 +38,57 @@ def average_rating
     def add_trip(trip)
       if trip.class != Trip
         raise ArgumentError.new("Can only add trip instance to trip collection")
+      elsif trip.end_time == nil
+      change_status
       end
-
       @trips << trip
     end
+
+    def change_status
+      if
+        @status == :UNAVAILABLE
+        @status = :AVAILABLE
+      else
+        @status = :UNAVAILABLE
+      end
+    end
+
+
+    def total_revenue
+      fee = 1.65
+      driver_take = 0.8
+
+
+      subtotal = 0
+      @trips.each do |trip|
+        if trip.cost == nil
+          next
+        else
+          subtotal += (trip.cost - fee)
+        end
+        # Question, what if the cost is less than the fee?
+      end
+
+      total = subtotal * driver_take
+      return total.round(2)
+    end
+
+    def average_revenue
+      revenue = total_revenue
+      hour_time = 0
+      hour = 60 * 60
+      trip_count = trips.length
+
+      trips.each do |trip|
+        if trip.end_time == nil
+          trip_count -= 1
+        else
```

Comment:
I see similar `if` statements to this one in several places in both `Driver` and `Passenger`. To DRY this up, it might work well to define a helper method that returns a list of completed trips:

```ruby
def completed_trips
  return @trips.reject{ |t| t.end_time.nil? }
end

def total_revenue
  completed_trips.each do |trip|
    # ... same logic as before ...
  end
end
```